experiment/refresh/server.go

```diff
@@ -163,17 +165,40 @@ func (s *Server) handleIssueComment(l *logrus.Entry, ic github.IssueCommentEvent
 		return nil
 	}
 
-	jenkinsReport := s.configAgent.Config().JenkinsOperator.ReportTemplate
+	jenkinsReports := s.configAgent.Config().JenkinsOperator
 	kubeReport := s.configAgent.Config().Plank.ReportTemplate
 	for _, pj := range pjutil.GetLatestProwJobs(presubmits, kube.PresubmitJob) {
-		s.log.WithFields(l.Data).Infof("Refreshing the status of job %q (pj: %s)", pj.Spec.Job, pj.ObjectMeta.Name)
-		reportTemplate := jenkinsReport
-		if pj.Spec.Agent == "kubernetes" {
+		var reportTemplate *template.Template
+		switch pj.Spec.Agent {
+		case kube.KubernetesAgent:
 			reportTemplate = kubeReport
+		case kube.JenkinsAgent:
+			if report := s.jobMatchesReport(&pj, jenkinsReports); report != nil {
+				reportTemplate = report
+			}
+		}
+		if reportTemplate == nil {
+			continue
 		}
+
+		s.log.WithFields(l.Data).Infof("Refreshing the status of job %q (pj: %s)", pj.Spec.Job, pj.ObjectMeta.Name)
 		if err := report.Report(s.ghc, reportTemplate, pj); err != nil {
 			s.log.WithError(err).WithFields(l.Data).Info("Failed report.")
 		}
 	}
 	return nil
 }
+
+func (s *Server) jobMatchesReport(pj *kube.ProwJob, configs []config.JenkinsOperator) *template.Template {
+	for _, cfg := range configs {
+		selector, err := labels.Parse(cfg.LabelSelector)
```

Comment:
Each label selector will be parsed for each of the prowjobs returned by `pjutil.GetLatestProwJobs(presubmits, kube.PresubmitJob)` instead of just being parsed once. Is this acceptable?

Reply:
I am already parsing these selectors during config load so I changed to also store the parsed selectors and simply consume here.